### PR TITLE
[REFACTOR] Deployment get command

### PR DESCRIPTION
**Pull request by dennisseah**
Related to https://github.com/microsoft/bedrock/issues/885

1. move command declaration to a JSON file
2. standardized on the function in command ts file
3. use the CommandBuilder's exit function
4. added more tests (hitting over 90% coverage)
5. resolve some ts-lint violations
6. declare interface for input values
7. added code to validate top option value - must be positive integer
8. modified `getDeployments` function to return `Promise` object
9. avoid calling `Config` function multiple times
10. reduce the number of parameters in functions

### CODE REVIEW

File: src/commands/deployment/get.ts [hunk 1]
```diff
@@ -2,95 +2,101 @@ import Table from "cli-table";
...
-   */
-  WIDE = 1,
+export interface IInitObject {
+  config: IConfigYaml;
+  clusterPipeline: AzureDevOpsPipeline;
+  hldPipeline: AzureDevOpsPipeline;
+  key: string;
+  srcPipeline: AzureDevOpsPipeline;
+}
+export interface ICommandOptions {
```
Comment (samiyaakhtar) [inline src/commands/deployment/get.ts:37]: nit: jsdoc for `ICommandOptions` and `IInitObject`

File: src/commands/deployment/get.ts [hunk 2]
```diff
@@ -2,95 +2,101 @@ import Table from "cli-table";
...
+    nTop: top,
+    output: opts.output,
+    outputFormat: processOutputFormat(opts.output),
+    service: opts.service,
+    top: opts.top,
+    watch: opts.watch
+  };
+};
+
+export const execute = async (
```
Comment (samiyaakhtar) [inline src/commands/deployment/get.ts:97]: nit: jsdoc for `execute`, `validateValues`

**Review by samiyaakhtar — APPROVED** [review]

File: src/commands/deployment/get.ts [hunk 3]
```diff
@@ -117,135 +123,108 @@ export const processOutputFormat = (outputFormat: string): OUTPUT_FORMAT => {
...
+      values.env,
+      values.imageTag,
+      values.buildId,
+      values.commitId,
+      values.service,
+      values.deploymentId
+    )
+      .then((deployments: Deployment[]) => {
+        if (values.outputFormat === OUTPUT_FORMAT.JSON) {
+          console.log(JSON.stringify(deployments, null, 2));
```
Comment (samiyaakhtar) [inline src/commands/deployment/get.ts:178]: Add `// tslint:disable-next-line: no-console`

File: src/commands/deployment/get.ts [hunk 4]
```diff
@@ -378,13 +356,8 @@ export const printDeployments = (
...
-      count++;
-      if (limit && count >= limit) {
-        break;
-      }
-    }
-    // Use console.log since it helps piping output to a file using > filename.txt
-    // tslint:disable-next-line: no-console
+    });
+
     console.log("\n" + table.toString());
```
Comment (samiyaakhtar) [inline src/commands/deployment/get.ts:390]: Same

File: src/commands/deployment/get.ts (hunk 1 above)
Comment (dennisseah) [reply src/commands/deployment/get.ts:37]: accept

File: src/commands/deployment/get.ts (hunk 2 above)
Comment (dennisseah) [reply src/commands/deployment/get.ts:97]: accept

File: src/commands/deployment/get.ts (hunk 3 above)
Comment (dennisseah) [reply src/commands/deployment/get.ts:178]: accept

File: src/commands/deployment/get.ts (hunk 4 above)
Comment (dennisseah) [reply src/commands/deployment/get.ts:390]: accept

**Comment by samiyaakhtar** [comment]
@dennisseah LGTM 👍 should we remove the log statement `--end log: 0 --`, or add a -f option? Let me know so I can unblock on integration test :)

**Comment by dennisseah** [comment]
> @dennisseah LGTM 👍 should we remove the log statement `--end log: 0 --`, or add a -f option? Let me know so I can unblock on integration test :)

@andrebriggs, can you chime in?

**Comment by samiyaakhtar** [comment]
I think we should get rid of `--end log: 0 --` or it becomes hard to parse the output from get command, we also just fixed a bug this morning around that. Thoughts? @andrebriggs

**Review by andrebriggs — APPROVED** [review]
LGTM
